src/audio-worklet/free-queue/src/interface/C/free_queue.h

```diff
@@ -0,0 +1,206 @@
+#ifndef FREE_QUEUE_C
+#define FREE_QUEUE_C
+
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdatomic.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef EMSCRIPTEN_KEEPALIVE
+#define EMSCRIPTEN_KEEPALIVE __attribute__((used))
+#endif
+
+/**
+ * C Struct to represent FreeQueue
```

Comment:
The comment should be meaningful and purposeful. Explain "why" and not "what". In this case, provide a general overview of this struct, and the goal of this struct.

Reply:
Also why can't this be a `class` with methods?

Reply:
I was having issues with embind, so then I thought of going C style, so it will work with both C and C++.
That's why had to go with struct without methods.